apache/superset
# feat(dashboard): confirm overwrite to prevent unintended changes

### SUMMARY
At Airbnb, there's reasonable amount of reporting about unintended overwritten of filter scopes settings.
We believe there must be unknown overwriting on json_metadata by other authors during editing. (json_metadata can be easily overwritten since it passes all attributes including unchanged values.)
This commit introduces the inspection step for this high volatility value changes and then proceed to update with the user's confirmation. (the inspection items are defined [here](https://github.com/apache/superset/pull/21819/files#diff-c070f1a491b40805e52139c971994f682921d2588cf1f06685d30ccf200cd419R44))
This commit also adds the logging for the user confirmation.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

https://user-images.githubusercontent.com/1392866/195903661-131177c7-812d-4a38-8d56-b6e429b27cbd.mov

### TESTING INSTRUCTIONS
- set `ENABLE_CONFIRM_OVERWRITE_DASHBOARD_METADATA` feature flag to true
- go to any dashboard page
- update css or/and filter scopes
- click save to confirm the overwrite values

### ADDITIONAL INFORMATION
- [x] Required feature flags: `ENABLE_CONFIRM_OVERWRITE_DASHBOARD_METADATA`

cc: @ktmud @john-bodley
